packages/services/src/view-service/transaction-planner/fees.ts

```diff
@@ -0,0 +1,34 @@
+import { AssetId } from '@buf/penumbra-zone_penumbra.bufbuild_es/penumbra/core/asset/v1/asset_pb';
+import { TransactionPlannerRequest } from '@buf/penumbra-zone_penumbra.bufbuild_es/penumbra/view/v1/view_pb';
+
+// Attempts to extract a fee token from the assets used in the actions of the planner request
+// Priority in descending order
+export const extractAltFee = (request: TransactionPlannerRequest): AssetId => {
+  const outputAsset = request.outputs.map(o => o.value?.assetId).find(Boolean);
+  if (outputAsset) return outputAsset;
+
+  const spendAsset = request.spends.map(o => o.value?.assetId).find(Boolean);
+  if (spendAsset) return spendAsset;
+
+  const swapAsset = request.swaps.map(assetIn => assetIn.value?.assetId).find(Boolean);
+  if (swapAsset) return swapAsset;
+
+  const auctionScheduleAsset = request.dutchAuctionScheduleActions
+    .map(a => a.description?.outputId)
+    .find(Boolean);
+  if (auctionScheduleAsset) return auctionScheduleAsset;
+
+  const auctionEndAsset = request.dutchAuctionEndActions.map(a => a.auctionId?.inner).find(Boolean);
+  if (auctionEndAsset) {
+    return new AssetId({ inner: auctionEndAsset });
+  }
```

Comment:
Hm... Auction IDs are not asset IDs. Are you thinking of an auction NFT? (Even if so, though, I can't imagine that gas fees could be paid with auction NFTs...?)

Reply:
Also, I'm noticing that all of the issues I've raised with this file were already present before this PR — you just moved the file in this PR. So, feel free to address those issues in a separate PR.

Reply:
that is indeed the unique id of the auction to close, which has me questioning what asset id would be relevant for the fee? think we'd need a database lookup here

Reply:
I think we should actually just not include them as targets for multi-fees in this pr, and handle them separately after more thought.

Reply:
see https://github.com/penumbra-zone/web/issues/1375 and https://github.com/penumbra-zone/web/issues/1414